cirq-web/cirq_ts/src/bloch_sphere/components/vector.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {VectorInput} from './types';
+import {ArrowHelper, Vector3, Group, LineBasicMaterial} from 'three';
+
+/**
+ * Generates a vector to add to the Bloch sphere. The length and direction of the vector are configurable.
+ */
+export class Vector extends Group {
+  readonly length: number;
+  readonly x: number;
+  readonly y: number;
+  readonly z: number;
+
+  /**
+   * Class constructor.
+   * @param vectorData A JSON string containing information used to build the vector.
+   * @returns An instance of the class containing the generated vector. This can be
+   * added to the Bloch sphere instance as well as the scene.
+   */
+  constructor(vector: VectorInput) {
+    super();
+    this.x = vector.x;
+    this.y = vector.y;
+    this.z = vector.z;
+    this.length = vector.length;
+
+    this.generateVector(this.x, this.y, this.z, this.length);
+    return this;
+  }
+
+  /**
+   * Generates a vector starting at (0, 0) and ending at the coordinates
+   * of the given parameters, and adds to group.
+   * Utilizes three.js ArrowHelper function generate the vector.
```

Comment:
These arguments are odd: x, y, z should unambiguously determine length of a vector just from the Pythagorean theorem.  

I understand that this came from the `cirq.bloch_vector_from_state_vector` as that gives the coordinates of a pure state on a sphere of radius 1, but that doesn't mean that it should influence the API in surprising ways.

Reply:
Three.js helps out with a distance calculation function..I removed code to normalize the vector and get the length of the vector based on the coordinates